lesson07/main.js

```diff
@@ -1,26 +1,48 @@
-new Promise((resolve)=> {
-  const elements = [ 
-    {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
-    {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
-  ]; 
-  setTimeout(() => {
-    resolve(elements);
-  }, 3000);
-}).then((resolvedElements)=> {
+
+function renderCircle(){
+  const loadingCircle = document.createElement("img"); 
+  const ul = document.getElementById("js-ul"); 
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "ローディング画像";
+  loadingCircle.id = "loading-circle";
+  ul.appendChild(loadingCircle);
+}  
+
+function removeCircle(){
+  document.getElementById("loading-circle").remove();
+}
+
+function renderMenus(resolvedMenus){

```

Comment:
renderMenusは渡って来た値が解決されたかどうかは知らないでもいいので
```suggestion
function renderMenus(values){

```